### PR TITLE
Add CircleCI integration to NOVA-Core

### PR DESCRIPTION
**Pull request by Temmie19**
I mean. That pretty much sums it up. This should be everything we need, but I'll happily fix it if it ever needs it.

### CODE REVIEW

**Comment by ExE-Boss** [comment]
You should [squash those commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing) and remove the change to `gradle.properties`.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - run: ./gradlew dependencies
+
+      - save_cache:
+          paths:
+            - ~/gradle
+          key: v1-dependencies-{{ checksum "build.gradle" }}
+        
+      # run tests!
+      - run: ./gradlew idea
+      - run: ./gradlew publishToMavenLocal
```
Comment (ExE-Boss) [inline .circleci/config.yml]: This can just be `- run: ./gradlew check`

Comment (ExE-Boss) [inline .circleci/config.yml]: This can just be `- run: ./gradlew check`

**Comment by Temmie19** [comment]
And that URL in the ``gradle.properties`` wasn't actually still there. That was removed in the next commit anyways.

**Comment by Temmie19** [comment]
So ``./gradlew check`` doesn't apprarently work.

**Comment by ExE-Boss** [comment]
That’s because of NOVA-Team/NOVA-Gradle#19.

The previous build worked because it didn’t run the tests.

**Pull request merged**